src/rgw/rgw_swift_auth.cc

```diff
@@ -210,8 +210,9 @@ class TempURLEngine::PrefixableSignatureHelper
       object_name(object_name),
       prefix(prefix) {
     /* Transform: v1/acct/cont/obj - > v1/acct/cont/ */
-    no_obj_uri = \
-      decoded_uri.substr(0, decoded_uri.length() - object_name.length());
+    const size_t no_obj_url_len = \
+      this->decoded_uri.length() - object_name.length();
+    no_obj_uri = this->decoded_uri.substr(0, no_obj_url_len);
```

Comment:
looks right. but renaming the constructor argument so it doesn't shadow the member variable may be preferable